models/models.js

```diff
@@ -48,25 +47,47 @@ exports.selectUsers = () => {
   });
 };
 
-exports.selectsArticles = () => {
-  return db
-    .query(
-      `SELECT users.username, 
-       articles.title, 
-       articles.article_id,
-       articles.topic,
-       articles.created_at,
-       articles.votes,
-       COUNT(comments.article_id) AS comment_count
-        FROM articles 
-          JOIN users ON articles.author = users.username
-     	    LEFT JOIN comments ON articles.article_id = comments.article_id 
-     	  GROUP BY articles.article_id, users.username
-       ORDER BY articles.created_at DESC;`
-    )
-    .then(({ rows: articles }) => {
-      return articles;
-    });
+exports.selectsArticles = (sortBy = "created_at", orderBy = "DESC", topic) => {
+  const validSortBys = [
+    "author",
+    "created_at",
+    "title",
+    "article_id",
+    "topic",
+    "votes",
+    "comment_count",
+  ];
+  const validOrderBys = ["desc", "DESC", "asc", "ASC"];
+
+  if (!validSortBys.includes(sortBy) || !validOrderBys.includes(orderBy)) {
+    return Promise.reject({ status: 400, msg: "Invalid input" });
+  }
+
+  let queryArr = [];
+  let queryStr = `SELECT articles.author, 
+  articles.title, 
+  articles.article_id,
+  articles.topic,
+  articles.created_at,
+  articles.votes,
+  CAST(COUNT(comments.article_id)AS INT) AS comment_count
+  FROM articles 
+  LEFT JOIN comments ON articles.article_id = comments.article_id 
+  GROUP BY articles.article_id`;
+
+  if (topic) {
+    queryStr += ` HAVING articles.topic = $1`;
+    queryArr.push(topic);
+  }
+
```

Comment:
We'll need to check if the topic exists as we don't want to send an error simply because we get no articles. If a topic exists but has not articles we should return an empty array